lamindb/curators/core.py

```diff
@@ -1761,48 +1902,57 @@ def annotate_artifact(
                 logger.important(
                     f"not annotating with {len(features)} features as it exceeds {settings.annotation.n_max_records} (ln.settings.annotation.n_max_records)"
                 )
-                itype = parse_cat_dtype(artifact.schema.itype, is_itype=True)["field"]
+                itype = (
+                    Feature.name
+                    if artifact.schema.itype == "Composite"
+                    else parse_cat_dtype(artifact.schema.itype, is_itype=True)["field"]
+                )
                 feature_set = Schema(itype=itype, n=len(features))
             artifact.feature_sets.add(
                 feature_set.save(), through_defaults={"slot": "columns"}
             )
     else:
-        for slot, slot_curator in curator._slots.items():
-            # var_index is backward compat (2025-05-01)
-            name = (
-                "var_index"
-                if (slot == "var" and "var_index" in slot_curator.cat._cat_vectors)
-                else "columns"
-            )
-            features = slot_curator.cat._cat_vectors[name].records
-            if features is None:
-                logger.warning(f"no features found for slot {slot}")
-                continue
-            validating_schema = slot_curator._schema
-            index_feature = validating_schema.index
-            feature_set = Schema(
-                features=[f for f in features if f != index_feature],
-                itype=validating_schema.itype,
-                index=index_feature,
-                minimal_set=validating_schema.minimal_set,
-                maximal_set=validating_schema.maximal_set,
-                coerce_dtype=validating_schema.coerce_dtype,
-                ordered_set=validating_schema.ordered_set,
-            )
-            if (
-                feature_set._state.adding
-                and len(features) > settings.annotation.n_max_records
-            ):
-                logger.important(
-                    f"not annotating with {len(features)} features for slot {slot} as it exceeds {settings.annotation.n_max_records} (ln.settings.annotation.n_max_records)"
+        if curator is not None:
```

Comment:
Why is this needed now but wasn't before? We already had generic logic that annotates slot curators. Why not just remove the special casing for the data frame curator that was necessary prior to it becoming a slot curator?

I don't like the high level of indendation this causes. Maybe I'm misunderstanding but I think curator should always be not None given we're in the curator module.

Reply:
This is actually a relict from before the refactoring. I missed this. Thanks! It's gone now.

Reply:
Great!